python/cudf/cudf/core/series.py

```diff
@@ -4022,7 +3994,7 @@ def _format_stats_values(stats_data):
 
         def describe_numeric(self):
             # mimicking pandas
-            names = (
+            index = (
```

Comment:
Not actually, the descibe functions have been separated primarily based on dtypes because each dtype have some variation in their index, i.e., numeric contain `"count", "mean", "std", "min", 25%, 50%, 75%, "max"` but categorical/string dtype contain `"count", "unique", "top", "freq"`. The datetime/time delta dtypes also differ in `std` too. 

Did you mean the avoiding of join to align them in dataframe.describe code? One thought I have is we could choose to   skip `names` generation by deterministic order if all dtypes of `data_to_describe` are of same types by querying `data_to_describe.dtypes`. Does that sound like a fair optimization?

Reply:
> Did you mean the avoiding of join to align them in dataframe.describe code? One thought I have is we could choose to skip `names` generation by deterministic order if all dtypes of `data_to_describe` are of same types by querying `data_to_describe.dtypes`. Does that sound like a fair optimization?

Sounds like a future optimization we can do if needed. `DataFrame.describe` is likely used in interactive fashion where the small join here won't matter.